### PR TITLE
feat(format/grit): grit formatter initial configuration

### PR DESCRIPTION
**Pull request by branberry**
## Summary

Create basic scaffolding/configuration for the GritQL formatter crate. I am realizing now that I probably started in the reverse order for the GritQL work, I probably should have started with the parser in hindsight. 

With that said, this gets the bulk of the setup with the formatter out of the way, minus some tests. Will work to add that next. 

#2476

## Test Plan

It build successfully.

### CODE REVIEW

File: crates/biome_formatter/CONTRIBUTING.md [hunk 1]
```diff
@@ -161,11 +161,11 @@ where
...
 
-Then, you'll have to create three types:
+Then, you'll have to create four types:
 1. `HtmlCommentStyle`
-1. `HtmlFormatContext`
-1. `FormatHtmlSyntaxNode` 
-1. `HtmlLanguage`
+2. `HtmlFormatContext`
+3. `FormatHtmlSyntaxNode` 
+4. `HtmlLanguage`
```
Comment (dyc3) [inline crates/biome_formatter/CONTRIBUTING.md:168]: nit: Actually, if you prefix these with `1.` it automatically numbers the items correctly when it renders.

**Review by dyc3 — COMMENTED** [review]
Looks good so far. Thanks for the doc tweaks too.

I've been working on the HTML formatter recently, so if you have any questions, feel free to ping me here on GitHub or on discord

File: crates/biome_formatter/CONTRIBUTING.md (hunk 1 above)
Comment (branberry) [reply crates/biome_formatter/CONTRIBUTING.md]: Ah, gotcha! Thanks for that. I wasn't aware of that, and I was only looking at it from the pure markdown, so I totally missed this

**Review by dyc3 — APPROVED** [review]

**Comment by dyc3** [comment]
Check CONTRIBUTING.md for our PR title format

**Comment by dyc3** [comment]
fyi You can run `just ready` to run the CI checks locally.

**Comment by branberry** [comment]
Thanks! That's exactly what I was looking for.

**Review by dyc3 — APPROVED** [review]

**Pull request merged**